### ISSUE TITLE
In Quarkus 3 platform or as external extension?

### ISSUE THREAD

**Issue by maxandersen** [datastax/cassandra-quarkus#230]
in https://github.com/datastax/cassandra-quarkus/pull/222#issuecomment-1493683166 its discussed having cassandra be ready for inclusion in Quarkus 3 platform - I would like to make sure that is the right thing or maybe it makes more sense cassandra becomes an external extension similar to i.e. neo4j is it or all the other quarkiverse projects.

Being in the platform should only be done if those maintaining cassandra-quarkus is able to participate and follow the regular updates in Quarkus platform. We've automated a lot of this but we know things can get busy and one might not be ready for inclusion all the time. 

Cassandra was early member of Quarkus platform since at the time we didn't have the notion of external extensions which is now fully supported in CLi and code.quarkus.io. We now have that and it would be good to consider what makes the most sense for cassandra?

@absurdfarce are you or someone else able to commit to check for regular updates in  Quarkus platform or would it make more sense to be an external extension?

┆Issue is synchronized with this [Jira Task](https://datastax-oss.atlassian.net/browse/QUAR-120) by [Unito](https://www.unito.io)

**Comment by maxandersen** [comment]
Yes. Im aware and much appreciated You updated to make the extension compatible with Quarkus 3. What Im asking and suggesting here different though. It is about if cassandra is included into Quarkus 3 platform or not. 

My suggestion is that it moves to be external extension similar to neo4j and other extensions from quarkiverse. 

Then cassandra and the platform Can Release Independently and doesnt have to align as coupled as done Thus far.

**Comment by absurdfarce** [comment]
@maxandersen There's still some internal discussion going on on this point but as of now my strong suspicion is that we'd like for the plugin to remain a part of Quarkus Platform.  Let me do some more checking internally, though, before I say that for real. :)

Thanks!

**Comment by maxandersen** [comment]
Sounds good. 

What I'll suggest is that for quarkus 3 platform we keep it out for now as better user experience if we don't add it until we can keep it there with a somewhat high certainty. 

Have you considered moving it to quarkiverse GitHub? That is way  others can more easily help and provide releases if need be.

**Comment by aloubyansky** [comment]
@absurdfarce we could have a short call to explain all the available options, find what suits your processes betters and explain what is expected from your participation in each case. Let us know in case it could help. Thanks.

**Comment by absurdfarce** [comment]
@aloubyansky Thanks, I think that might be a good idea.  I'd like to at least vet my understanding of the additional work required to support the Quarkus Platform in order to make sure I understand the rough size of what's involved; I have a pretty good idea now but I'd like to be very sure.

I'd also like to invite one of our product managers along so that they can be aware of what we're committing to from that angle.

**Comment by aloubyansky** [comment]
@absurdfarce I sent you a message on LinkedIn asking for your email :)

**Comment by absurdfarce** [comment]
@aloubyansky Apologies, hadn't replied to this yet... but fortunately I just did :)

**Comment by absurdfarce** [comment]
My understanding is that we've come to a resolution on this matter so I'm closing this out.  If we need to revisit we can do so!
